### ISSUE TITLE
Error: GitHub Failure: Bad credentials

### ISSUE THREAD

**Issue by OliverTetzTT** [renovatebot/renovate#17153]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.151.3

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.5.3

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Hi folks,

We have Renovate running in our CI using the official docker image. Everything was fine until a few days ago. Now renovate has problems when trying to authenticate to github.com. We already tried multiple Access Tokens but none is working so far. 

When I rebuild this request by the information given in the Log entry, I don't get the 401 error code. 

Maybe there is another solution we didn't consider.

Thanks :) 

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GitHub failure: Bad credentials
       "token": "***********",
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1660295633390,
           "socket": 1660295633392,
           "lookup": 1660295633482,
           "connect": 1660295633490,
           "secureConnect": 1660295633514,
           "upload": 1660295633515,
           "response": 1660295633635,
           "end": 1660295633638,
           "phases": {
             "wait": 2,
             "dns": 90,
             "tcp": 8,
             "tls": 24,
             "request": 1,
             "firstByte": 120,
             "download": 3,
             "total": 248
           }
         },
         "message": "Response code 401 (Unauthorized)",
         "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/32.151.3 (https://github.com/renovatebot/renovate)",
             "accept": "application/vnd.github.v3+json",
             "authorization": "***********",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://api.github.com/user",
           "hostType": "github",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
         "response": {
           "statusCode": 401,
           "statusMessage": "Unauthorized",
           "body": {
             "message": "Bad credentials",
             "documentation_url": "https://docs.github.com/rest"
           },
           "headers": {
             "server": "GitHub.com",
             "date": "Fri, 12 Aug 2022 09:13:53 GMT",
             "content-type": "application/json; charset=utf-8",
             "content-length": "80",
             "x-github-media-type": "github.v3; format=json",
             "x-ratelimit-limit": "60",
             "x-ratelimit-remaining": "51",
             "x-ratelimit-reset": "1660296154",
             "x-ratelimit-used": "9",
             "x-ratelimit-resource": "core",
             "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
             "access-control-allow-origin": "*",
             "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
             "x-frame-options": "deny",
             "x-content-type-options": "nosniff",
             "x-xss-protection": "0",
             "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
             "content-security-policy": "default-src 'none'",
             "vary": "Accept-Encoding, Accept, X-Requested-With",
             "x-github-request-id": "4EC0:8326:52CE5:56D1D:62F619D1",
             "connection": "close"
           },
           "httpVersion": "1.1"
         }
       }
DEBUG: Error authenticating with GitHub
       "err": {
         "hostType": "github",
         "err": {
           "name": "HTTPError",
           "code": "ERR_NON_2XX_3XX_RESPONSE",
           "timings": {
             "start": 1660295633390,
             "socket": 1660295633392,
             "lookup": 1660295633482,
             "connect": 1660295633490,
             "secureConnect": 1660295633514,
             "upload": 1660295633515,
             "response": 1660295633635,
             "end": 1660295633638,
             "phases": {
               "wait": 2,
               "dns": 90,
               "tcp": 8,
               "tls": 24,
               "request": 1,
               "firstByte": 120,
               "download": 3,
               "total": 248
             }
           },
           "message": "Response code 401 (Unauthorized)",
           "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
           "options": {
             "headers": {
               "user-agent": "RenovateBot/32.151.3 (https://github.com/renovatebot/renovate)",
               "accept": "application/vnd.github.v3+json",
               "authorization": "***********",
               "accept-encoding": "gzip, deflate, br"
             },
             "url": "https://api.github.com/user",
             "hostType": "github",
             "username": "",
             "password": "",
             "method": "GET",
             "http2": false
           },
           "response": {
             "statusCode": 401,
             "statusMessage": "Unauthorized",
             "body": {
               "message": "Bad credentials",
               "documentation_url": "https://docs.github.com/rest"
             },
             "headers": {
               "server": "GitHub.com",
               "date": "Fri, 12 Aug 2022 09:13:53 GMT",
               "content-type": "application/json; charset=utf-8",
               "content-length": "80",
               "x-github-media-type": "github.v3; format=json",
               "x-ratelimit-limit": "60",
               "x-ratelimit-remaining": "51",
               "x-ratelimit-reset": "1660296154",
               "x-ratelimit-used": "9",
               "x-ratelimit-resource": "core",
               "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
               "access-control-allow-origin": "*",
               "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
               "x-frame-options": "deny",
               "x-content-type-options": "nosniff",
               "x-xss-protection": "0",
               "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
               "content-security-policy": "default-src 'none'",
               "vary": "Accept-Encoding, Accept, X-Requested-With",
               "x-github-request-id": "4EC0:8326:52CE5:56D1D:62F619D1",
               "connection": "close"
             },
             "httpVersion": "1.1"
           }
         },
         "message": "external-host-error",
         "stack": "Error: external-host-error\n    at handleGotError (/usr/src/app/node_modules/renovate/lib/util/http/github.ts:127:14)\n    at GithubHttp.request (/usr/src/app/node_modules/renovate/lib/util/http/github.ts:336:13)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at GithubHttp.requestJson (/usr/src/app/node_modules/renovate/lib/util/http/index.ts:219:17)\n    at getUserDetails (/usr/src/app/node_modules/renovate/lib/modules/platform/github/user.ts:13:7)\n    at Proxy.initPlatform (/usr/src/app/node_modules/renovate/lib/modules/platform/github/index.ts:145:36)\n    at initPlatform (/usr/src/app/node_modules/renovate/lib/modules/platform/index.ts:46:24)\n    at globalInitialize (/usr/src/app/node_modules/renovate/lib/workers/global/initialize.ts:63:12)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:119:14)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
FATAL: Authentication failure
DEBUG: Renovate exiting
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "J9SpLhtRRJc9uw3-VxMqd",
           "msg": "Authentication failure"
         }
       ]

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by PhilipAbed** [comment]
you seem to be the only one having the problem, otherwise people will go crazy over this, lots of github users out there,
since you didn't provide any data, and i cant see in history any changes for github,

can you please trouble shoot it for us?
i believe that if it was working and then it stopped, then it is supposed to work on older renovate versions right?

Troubleshooting:
Try running on older versions, see if the problem is really in the latest releases, 
if yes, you can search for when it started happening, making it easier for us to resolve the issue.

**Comment by PhilipAbed** [comment]
also it would be helpful to see if your config is actually being read, maybe show us logs where your config is printed, (redact sensitive data ofcourse)

**Comment by nabeelsaabna** [comment]
failing when calling `https://api.github.com/user`
please double check the token

**Comment by viceice** [comment]
maybe they use an app token instead of a pat? then they need to set username

**Comment by OliverTetzTT** [comment]
**TL;DR:** The config wasn't read as it was not mounted correctly. As this wasn't right, the wrong token was used. The mounting broke due to some setup change. Getting the config.js back to the right spot fixed the problem instantly.

**Long version:**

> also it would be helpful to see if your config is actually being read, maybe show us logs where your config is printed, (redact sensitive data ofcourse)

Thank you very much, for helping out. The quoted comment of @PhilipAbed helped by pushing me towards the underlying issue. It wasn't loading the config correctly (and therefore not using the correct token and so on). (I'll learn from this to include the full log instead of cutting it. I promise)

**Why did it work before?**

Some Docker magic. We set up a Docker runner of GitHub Action Runners to run a Renovate Container. 

With this we mounted the config at the expected path. At first everything worked fine. But after some time it broke. I suppose Docker daemon still had cached or stored some old volume and was using this. After clean up or executing on another machine, this didn't work anymore. That caching thingy is probably the reason, why we didn't pin point the moment our renovate stopped working to moving to the docker runner setup.

Our method isn't really suited to run Docker with it's full feature set as mounting a file from a docker container to another docker container does not work (if I read correctly, it creates an empty folder somewhere and tries to mount this -> resulting in it, of course, not working).

**Current Workaround**

We built the config file into the Container image. This wasn't as much of a problem, as we already built our own renovate image to install certificates. This works for now until we have a better idea. 


Hope this answer maybe helps some one else in the future. I'll close this issue and again thank you for the hints.
